File: 2471-minimum-amount-of-time-to-collect-garbage/minimum-amount-of-time-to-collect-garbage.py
class Solution:
    def garbageCollection(self, garbage: List[str], travel: List[int]) -> int:
        g, p, m = 0, 0, 0
        tp, tg, tm = 0, 0, 0

        for i in range(len(garbage)):
            for c in garbage[i]:
                if c == 'P':
                    p += tp + 1
                    tp = 0
                elif c == 'G':
                    g += tg + 1
                    tg = 0
                else:
                    m += tm + 1
                    tm = 0
            if i == len(travel): break
            tp += travel[i]
            tg += travel[i]
            tm += travel[i]
        
        return p + g + m

File: 2471-minimum-amount-of-time-to-collect-garbage/minimum-amount-of-time-to-collect-garbage.java
class Solution {
    public int garbageCollection(String[] garbage, int[] travel) {
        int g = 0, p = 0, m = 0;
        int tg = 0, tp = 0, tm = 0;
        for(int i=0; i<garbage.length; i++) {
            for(char ch : garbage[i].toCharArray()) {
                if(ch == 'M') {
                    m += tm;
                    m++;
                    tm = 0;
                }
                else if(ch == 'G') {
                    g += tg;
                    g++;
                    tg = 0;
                }
                else {
                    p += tp;
                    p++;
                    tp = 0;
                }
            }
            if(i == travel.length) break;
            tg += travel[i];
            tm += travel[i];
            tp += travel[i];
        }
        return g + p + m;
    }
}